devito/finite_differences/coefficients.py

```diff
@@ -0,0 +1,256 @@
+import sympy
+import numpy as np
+from cached_property import cached_property
+
+from devito.finite_differences import generate_indices
+from devito.tools import filter_ordered
+
+__all__ = ['Coefficient', 'CoefficientRules', 'default_rules']
+
+
+class Coefficient(object):
+    """
+    Prepare custom FD coefficients to pass to CoefficientRules object.
```

Comment:
for generality, I'd drop FD? wdy think?

Reply:
Sure. There are various 'coefficients' being named elsewhere in the code base (i.e. Sparse functions have a `_coefficients` property) so not sure how much we want to distinguish to avoid confusion? What does @mloubout think?